src/SME.SGP.Aplicacao/Queries/PlanoAEE/ObterPlanosAEE/ObterPlanosAEEQueryHandler.cs

```diff
@@ -28,12 +29,33 @@ public async Task<PaginacaoResultadoDto<PlanoAEEResumoDto>> Handle(ObterPlanosAE
             var usuario = await mediator.Send(ObterUsuarioLogadoQuery.Instance);
             bool ehAdmin = usuario.EhAdmGestao();
             bool ehPAEE = usuario.EhProfessorPaee();
-            var turmasCodigos = await ObterCodigosTurmas(request.UeId, ehAdmin);
+            string[] turmasCodigos = null;
+            long[] ues = new long[] { request.UeId };
+
+            if (request.UeId == 0 || request.UeId == -99)
+            {
+                ues = await ObterUesPorUsuario(request.DreId, request.AnoLetivo, request.ConsideraHistorico, usuario);
```

Comment:
Se o resultado de ues for null, aqui já irá descartar a ue inicializada na linha 33. O ideal para agrupar as ues do usuário, é um concat tratando antes se não é nulo, e depois até pode ter um distinct para não houver repetições. O que acha?